### PR TITLE
ROX-18480: rename feature flag

### PR DESCRIPTION
**Pull request by stehessel**
## Description

We decided to rename the feature from "Central Events" to "Central Notifications". This change renames the feature flag and sets up a deployment toggle.

### CODE REVIEW

File: deploy/common/env.sh [hunk 1]
```diff
@@ -66,6 +66,8 @@ echo "ROX_HOTRELOAD is set to ${ROX_HOTRELOAD}"
 export ROX_DECLARATIVE_CONFIGURATION="${ROX_DECLARATIVE_CONFIGURATION:-}"
 echo "ROX_DECLARATIVE_CONFIGURATION is set to ${ROX_DECLARATIVE_CONFIGURATION}"
 
+echo "ROX_CENTRAL_NOTIFICATIONS is set to ${ROX_CENTRAL_NOTIFICATIONS}"
```
Comment (rukletsov) [inline deploy/common/env.sh:69]: I think it would be great to stick to a single name of "the thing" in schema, service, env var, etc. (In previous PR "the thing" is called just "notifications".)

File: pkg/notifications/stream.go [hunk 2]
```diff
@@ -12,7 +12,8 @@ type Stream interface {
 	Produce(event *storage.Notification) error
```
Comment (rukletsov) [inline pkg/notifications/stream.go:12]: `event`?

File: pkg/notifications/stream.go [hunk 3]
```diff
@@ -12,7 +12,8 @@ type Stream interface {
...
 }
 
-func newStream() Stream {
+// NewStream creates a new notification stream.
+func NewStream() Stream {
```
Comment (rukletsov) [inline pkg/notifications/stream.go:16]: I guess I'll the reason in the following PRs but here it's unclear why `newStream()` should become exported. Please strive to colocate relevant changes in one PR.

File: pkg/notifications/stream.go (hunk 2 above)
Comment (stehessel) [reply pkg/notifications/stream.go:12]: Eh missed the rename there.

File: pkg/notifications/stream.go [hunk 4]
```diff
@@ -24,13 +25,19 @@ type streamImpl struct {
 
 // Consume returns the channel to retrieve notifications.
 func (s *streamImpl) Consume() <-chan *storage.Notification {
+	if s == nil {
+		return nil
+	}
```
Comment (dhaus67) [inline pkg/notifications/stream.go:30]: I suppose these changes (along with the ones for `Produce`) should be part of another PR to explicitly handle the nil case?

**Review by pedrottimark — APPROVED** [review]
UI review for 3 out of 8 files:
* pkg/features/list.go
* tests/e2e/lib.sh
* ui/apps/platform/src/types/featureFlag.ts

**Review by dhaus67 — APPROVED** [review]

**Comment by dhaus67** [comment]
For reviewers: taking over this PR. Since I'll likely make a commit to address some things here, I'll dismiss my review.

File: pkg/administration/events/converter.go [hunk 5]
```diff
@@ -4,7 +4,7 @@ import (
 	"github.com/stackrox/rox/generated/storage"
 )
 
-// LogConverter converts a log entry to a storage.AdministrationEvent.
+// LogConverter converts a log entry to an storage.AdministrationEvent.
```
Comment (rukletsov) [inline pkg/administration/events/converter.go:7]: I don't mind much any way, but why?

**Review by rukletsov — APPROVED** [review]

File: pkg/notifications/stream.go (hunk 3 above)
Comment (dhaus67) [reply pkg/notifications/stream.go]: Reverted the change, it only was required because it was used in a test in a later PR.

Refactored it to use a separate exported function like `GetTestStream(t *testing.T)` to only allow public usage in tests.
